### DISCUSSION TITLE
How to do: do not load video before play-button is pressed?

### DISCUSSION THREAD

**Discussion by oioix** [sampotts/plyr#2200] [Q&A]
Hi,
I love this player. And tried it on my homepage.
But I realised, that the page needs long time to load, because it does start loading the video (100MB). Vistitors will think, the page ist not working.
Beside of this it needs tons of bandwith, even if people ot watching the video.

To mention: the source is to deliver from my server. I do not want to use youtube or vimeo!

My question:
Is there any way, that the video file load does not start before the play-button is pressed?

Thanks in advance

**Comment by Sananthu-47** [comment]
First set the video tag with src='' and then add this functionallity
`player.on('play',function(){
                player.source = {
                    type: 'video',
                    title: 'Title of video',
                    sources: [
                        {
                        src: 'Place your source here',
                        type: 'video/mp4'
                        }
                    ],
                    poster: 'Place thumbnail here'
                    };
            });`

If you want the poster to be visible before the video play add `poster='Place the thumbnail here' ` to the video-tag

Hope this helps

**Comment by fekle** [comment]
You can also address this by simply setting the `preload` attribute on the HTML video element itself. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#attr-preload.
